master/internal/db/postgres_experiments.go

```diff
@@ -1117,22 +1117,31 @@ FROM (
 }
 
 // AddExperiment adds the experiment to the database and sets its ID.
-func (db *PgDB) AddExperiment(experiment *model.Experiment) error {
+func (db *PgDB) AddExperiment(experiment *model.Experiment) (err error) {
 	if experiment.ID != 0 {
 		return errors.Errorf("error adding an experiment with non-zero id %v", experiment.ID)
 	}
-	err := db.namedGet(&experiment.ID, `
-INSERT INTO experiments
-(state, config, model_definition, start_time, end_time, archived,
- git_remote, git_commit, git_committer, git_commit_date, owner_id, original_config, notes)
-VALUES (:state, :config, :model_definition, :start_time, :end_time, :archived,
-		:git_remote, :git_commit, :git_committer, :git_commit_date, :owner_id, :original_config,
-		:notes)
-RETURNING id`, experiment)
-	if err != nil {
-		return errors.Wrapf(err, "error inserting experiment %v", *experiment)
-	}
-	return nil
+	return db.withTransaction("add_experiment", func(tx *sqlx.Tx) error {
+		job := model.Job{
+			JobID:   experiment.JobID,
+			JobType: "EXPERIMENT",
+		}
+		if err = addJob(tx, &job); err != nil {
+			return errors.Wrapf(err, "error inserting job %v", job)
+		}
+		err := namedGet(tx, &experiment.ID, `
+	INSERT INTO experiments
+	(state, config, model_definition, start_time, end_time, archived,
+	 git_remote, git_commit, git_committer, git_commit_date, owner_id, original_config, notes, job_id)
```

Comment:
let's go ahead and add `job_id` to the selects too, so experiment queries don't return "" job_id's

Reply:
specifically like ExperimentByID and friends

Reply:
copilot in action:

![image](https://user-images.githubusercontent.com/12127420/139511534-601a8cac-9ae4-4800-8eac-9a5974e7d86f.png)

Reply:
incredible. i'm scared, it's official

Reply:
I could also add it to our old endpoint db helpers eg the ones here https://github.com/determined-ai/determined/blob/41b20e6680fe7eb5fe9f5ec55cce8ee361f7f604/master/internal/core_experiment.go#L78 but do we keep updating those? or should we let them fall off

Reply:
e.name is missing from a bunch of these as well. I'll avoid adding them to avoid adding more risk.